dbms/src/Functions/tests/gtest_datetime_extract.cpp

```diff
@@ -39,73 +36,48 @@ class TestDateTimeExtract : public DB::tests::FunctionTest
 {
 };
 
-// Disabled for now, since we haven't supported ExtractFromString yet
-TEST_F(TestDateTimeExtract, DISABLED_ExtractFromString)
+TEST_F(TestDateTimeExtract, ExtractFromString)
 try
 {
-    const Context context = TiFlashTestEnv::getContext();
-
-    auto & factory = FunctionFactory::instance();
-
     std::vector<String> units{
-        "year",
-        "quarter",
-        "month",
-        "week",
-        "day",
         "day_microsecond",
         "day_second",
         "day_minute",
         "day_hour",
-        "year_month",
     };
     String datetime_value{"2021/1/29 12:34:56.123456"};
```

Comment:
more tests with white spaces in the front or the tail of the string?

Reply:
sure